src/masternodes/mn_checks.cpp

```diff
@@ -1029,6 +1079,406 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
+                return Res::Err("amount in ext htlc must match the amount in the offer - %s != %s!", ValueFromAmount(submitexthtlc.amount).getValStr(), ValueFromAmount(offer->amount).getValStr());
+            
+            bool found=false;
+            mnview.ForEachICXSubmitEXTHTLC([&found,&submitexthtlc](CICXOrderView::StatusTxidKey const & key, uint8_t i) {
+                if (key.first.second != submitexthtlc.offerTx)
+                    return false;
+                found=true;
+                return false;
+            },{CICXSubmitDFCHTLC::STATUS_OPEN,submitexthtlc.offerTx});
+            if (found)
+                return Res::Err("ext htlc already submitted!");
+
+            std::unique_ptr<CICXSubmitDFCHTLCImplemetation> dfchtlc;
+            mnview.ForEachICXSubmitDFCHTLC([&](CICXOrderView::StatusTxidKey const & key, uint8_t) {
+                if (key.first.second == submitexthtlc.offerTx)
+                {
+                    dfchtlc = mnview.GetICXSubmitDFCHTLCByCreationTx(key.second);
+                    return (false);
+                }
+                return false;
+            },{CICXSubmitDFCHTLC::STATUS_OPEN,submitexthtlc.offerTx});
+            if (!dfchtlc)
+                return Res::Err("offer %s needs to have dfc htlc submitted first, but no dfc htlc found!", submitexthtlc.offerTx.GetHex());
+
+            if (submitexthtlc.hash != dfchtlc->hash)
+                return Res::Err("Invalid hash, external htlc hash is different than dfc htlc hash! %s != %s",submitexthtlc.hash.GetHex(),dfchtlc->hash.GetHex());
+        }
+        else
+        {
+            if (submitexthtlc.amount != offer->amount*order->orderPrice/COIN)
+                return Res::Err("amount in dfc htlc must match the amount necessary for offer amount - %s != %s!", ValueFromAmount(submitexthtlc.amount).getValStr(), ValueFromAmount(offer->amount*order->orderPrice/COIN).getValStr());
+
+            bool found=false;
+            mnview.ForEachICXSubmitEXTHTLC([&found,&submitexthtlc](CICXOrderView::StatusTxidKey const & key, uint8_t i) {
+                if (key.first.second != submitexthtlc.offerTx)
+                    return false;
+                found=true;
+                return false;
+            },{CICXSubmitDFCHTLC::STATUS_OPEN,submitexthtlc.offerTx});
+            if (found)
+                return Res::Err("ext htlc already submitted!");
+
+            found=false;
+            mnview.ForEachICXSubmitDFCHTLC([&found,&submitexthtlc](CICXOrderView::StatusTxidKey const & key, uint8_t) {
+                if (key.first.second == submitexthtlc.offerTx)
+                {
+                    found = true;
+                    return (false);
+                }
+                return true;
+            },{CICXSubmitDFCHTLC::STATUS_OPEN,offer->creationTx});
+            if (found)
+                return Res::Err("offer %s needs to have dfc htlc submitted first, but external htlc already submitted!", submitexthtlc.offerTx.GetHex());
```

Comment:
@Mixa84 ok must be copy-paste error here then

Reply:
You can see that two texts are different.
`offer %s needs to have dfc htlc submitted first, but external htlc already submitted!`
`offer %s needs to have dfc htlc submitted first, but no dfc htlc found!`
One (on line 1367) says it needs DFC HTLC first bout found EXT first, and the other (on line 1337) it needs DFC HTLC first but no DFC found!

Reply:
I need more clarity on this one then:

`offer %s needs to have dfc htlc submitted first, but external htlc already submitted!`

How can external already be submitted when it needs dfc first?